frontend/src/components/Analysis/AnalysisReport/index.tsx

```diff
@@ -19,140 +22,379 @@ interface Props {
   task: string;
   systems: SystemModel[];
   singleAnalyses: SystemAnalysesReturn["single_analyses"];
-  metricToSystemAnalysesParsed: MetricToSystemAnalysesParsed;
-  featureKeyToBucketInfo: FeatureKeyToUIBucketInfo;
-  updateFeatureKeyToBucketInfo: (
-    featureKey: string,
-    bucketInfo: UIBucketInfo
+  metricToSystemAnalysesParsed: {
+    [metric: string]: { [feature: string]: ResultFineGrainedParsed[] };
+  };
+  featureNameToBucketInfo: { [feature: string]: BucketIntervals };
+  updateFeatureNameToBucketInfo: (
+    featureName: string,
+    bucketInfo: BucketIntervals
   ) => void;
 }
 
-export function AnalysisReport(props: Props) {
-  const {
-    task,
-    systems,
-    metricToSystemAnalysesParsed,
-    featureKeyToBucketInfo,
-    updateFeatureKeyToBucketInfo,
-  } = props;
-  const systemIDs = systems.map((system) => system.system_id);
-  const systemNames = systems.map((system) => system.system_info.model_name);
-  const metricNames = Object.keys(metricToSystemAnalysesParsed);
-  const [activeMetric, setActiveMetric] = useState<string>(metricNames[0]);
-  const [activeSystemExamples, setActiveSystemExamples] =
-    useState<ActiveSystemExamples>();
-  // page number of the analysis table
-  const [page, setPage] = useState(0);
-
-  let maxRightBoundsLength = 0;
-  for (const bucketInfo of Object.values(featureKeyToBucketInfo)) {
-    maxRightBoundsLength = Math.max(
-      maxRightBoundsLength,
-      bucketInfo.rightBounds.length
-    );
-  }
-
+export function getColSpan(props: Props) {
```

Comment:
Some functions like this are only used internally, so we don't need to export them.

Reply:
Ahh, good point @OscarWang114 ! I removed the exports on the functions that didn't need to be exported.